File: apps/android/app/src/main/java/app/yula/android/data/api/ApiResponse.kt
package app.yula.android.data.api

import app.yula.android.data.models.User

data class ApiResponse<T>(
    val success: Boolean,
    val data: T? = null,
    val error: ApiError? = null
)

data class ApiError(
    val code: String,
    val message: String
)

data class AuthResponse(
    val user: User,
    val accessToken: String
)

data class PaginatedResponse<T>(
    val items: List<T>,
    val total: Int,
    val page: Int,
    val pageSize: Int,
    val hasMore: Boolean
)

File: apps/ios/YULA/Models/APIResponse.swift
import Foundation

struct APIResponse<T: Codable>: Codable {
    let success: Bool
    let data: T?
    let error: APIError?
}

struct APIError: Codable {
    let code: String
    let message: String
}

struct PaginatedResponse<T: Codable>: Codable {
    let items: [T]
    let total: Int
    let page: Int
    let pageSize: Int
    let hasMore: Bool
}

struct AuthResponse: Codable {
    let user: User
    let accessToken: String
}
